manager/integration/tests/test_basic.py

```diff
@@ -708,6 +709,243 @@ def test_backup_block_deletion(set_random_backupstore, client, core_api, volume_
     delete_backup_volume(client, volume_name)
 
 
+def test_dr_volume_with_backup_block_deletion(set_random_backupstore, client, core_api, volume_name):  # NOQA
+    """
+    Test DR volume last backup after block deletion.
+
+    Context:
+
+    We want to make sure that when the block is delete, the DR volume picks up
+    the correct last backup.
+
+    Steps:
+
+    1.  Create a volume and attach to the current node.
+    2.  Write 4 MB to the beginning of the volume (2 x 2MB backup blocks).
+    3.  Create backup(0) of the volume.
+    4.  Overwrite backup(0) blocks of data on the volume.
+    5.  Create backup(1) of the volume.
+    6.  Verify backup block count == 3.
+    7.  Create DR volume from backup(1).
+    8.  Verify DR volume last backup is backup(1).
+    9.  Delete backup(1).
+    10. Verify backup block count == 2.
+    11. Verify DR volume last backup is backup(0).
+    12. Overwrite backup(0) blocks of data on the volume.
```

Comment:
Overwrite the 1st block of data on the volume.
Same below.

Reply:
Thanks @shuo-wu , this is addressed.